.teamcity/generated/ImageValidation.kts

```diff
@@ -6,116 +6,90 @@ package generated
-        triggers {
-            finishBuildTrigger {
-                buildType = "${PublishHubVersion.publish_hub_version.id}"
-            }
-        }
-
-        requirements {
-            // -- compatibility with Windows images
-            contains("teamcity.agent.jvm.os.name", "Windows")
-        }
-
-        features {
-            dockerSupport {
-                cleanupPushedImages = true
-                loginToRegistry = on {
-                    dockerRegistryId = "PROJECT_EXT_774,PROJECT_EXT_315"
-                }
-            }
-        }
-//	dependencies {
-//		 dependency(AbsoluteId("TC_Trunk_DockerImages_push_hub_windows")) {
-//			 snapshot { onDependencyFailure = FailureAction.ADD_PROBLEM }
-//		 }
-//		 dependency(AbsoluteId("TC_Trunk_DockerImages_push_hub_linux")) {
-//			 snapshot { onDependencyFailure = FailureAction.ADD_PROBLEM }
-//		 }
-
-        // -- build number dependency
-//        dependency(AbsoluteId("TC_Trunk_BuildDistDocker")) {
-//            snapshot {
-//                reuseBuilds = ReuseBuilds.ANY
-//                onDependencyFailure = FailureAction.IGNORE
-//            }
-//        }
-//	}
-    })
+object image_validation: BuildType({
+
+	name = "Validation of Size Regression - Staging Docker Images (Windows / Linux)"
+	buildNumberPattern="test-%build.counter%"
+
+	vcs {
+		root(TeamCityDockerImagesRepo.TeamCityDockerImagesRepo)
+	}
+
+	triggers {
+		// Execute the build once the images are available within %deployRepository%
+		finishBuildTrigger {
+			buildType = "${PublishLocal.publish_local.id}"
+		}
+	}
+
+	params {
+		// Inherited parameter, not used within build configuration, thus removed in order to ...
+		// ... prevent unnecessary dependency.
+		param("dockerImage.teamcity.buildNumber", "-")
+	}
+
+	val images = listOf("%docker.deployRepository%teamcity-agent%docker.buildImagePostfix%:2022.10-windowsservercore-1809",
+		"%docker.deployRepository%teamcity-agent%docker.buildImagePostfix%:2022.10-nanoserver-1809",
+		"%docker.deployRepository%teamcity-minimal-agent%docker.buildImagePostfix%:2022.10-nanoserver-1809",
+		"%docker.deployRepository%teamcity-server%docker.buildImagePostfix%:2022.10-nanoserver-2004",
+		"%docker.deployRepository%teamcity-agent%docker.buildImagePostfix%:2022.10-windowsservercore-2004",
+		"%docker.deployRepository%teamcity-agent%docker.buildImagePostfix%:2022.10-nanoserver-2004",
+		"%docker.deployRepository%teamcity-minimal-agent%docker.buildImagePostfix%:2022.10-nanoserver-2004",
+		// -- linux images
+		"%docker.deployRepository%teamcity-agent%docker.buildImagePostfix%:2022.10-linux",
+		"%docker.deployRepository%teamcity-agent%docker.buildImagePostfix%:2022.10-linux-sudo",
+		"%docker.deployRepository%teamcity-minimal-agent%docker.buildImagePostfix%:2022.10-linux"
+	)
+
+	steps {
+		images.forEach { imageFqdn ->
+			// Generate validation for each image fully-qualified domain name (FQDN)
+			gradle {
+				name = "Image Verification Gradle - $imageFqdn"
```

Comment:
This parameter isn't resolved, and a step name looks like 
`Image Verification Gradle - %docker.deployRepository%teamcity-server%docker.buildImagePostfix%:2022.10-nanoserver-2004`

Reply:
@burnasheva it seems that static fields inside of Build configuration, such as the name of the step, aren't resolved during compile time. I think it is expected, as it's converted into XML within TeamCity Server.
I've addressed the change in https://github.com/JetBrains/teamcity-docker-images/pull/46